drivers/rtc/rtc_smartbond.c

```diff
@@ -0,0 +1,632 @@
+	}
+
+	return mask;
+}
+
+static int rtc_smartbond_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+						const struct rtc_time *timeptr)
+{
+	int ret = 0;
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	__unused uint32_t rtc_time_alarm_reg;
+	__unused uint32_t rtc_calendar_alarm_reg;
+	__unused uint32_t rtc_alarm_enable_reg;
+	__unused uint32_t rtc_status_reg;
+
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	if (mask & ~RTC_SMARTBOND_SUPPORTED_ALARM_FIELDS) {
+		LOG_ERR("Invalid alarm mask");
+		return -EINVAL;
+	}
+
+	if ((timeptr == NULL) && (mask != 0)) {
+		LOG_ERR("No pointer is provided to set alarm");
+		return -EINVAL;
+	}
+
+	if (!data->is_rtc_configured) {
+		LOG_WRN("RTC is not initialized yet");
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	rtc_alarm_enable_reg = RTC->RTC_ALARM_ENABLE_REG;
+
+	/* Disable alarm to obtain coherency and/or when the alarm mask is empty */
+	RTC->RTC_ALARM_ENABLE_REG = 0;
+	RTC->RTC_INTERRUPT_DISABLE_REG = RTC_RTC_INTERRUPT_DISABLE_REG_RTC_ALRM_INT_DIS_Msk;
+
+	if (mask) {
+		/* Store current counter values as it might happen requested alrm is not valid */
+		rtc_time_alarm_reg = RTC->RTC_TIME_ALARM_REG;
+		rtc_calendar_alarm_reg = RTC->RTC_CALENDAR_ALARM_REG;
+
+		RTC->RTC_TIME_ALARM_REG =  alarm_time_to_bcd(timeptr);
+		RTC->RTC_CALENDAR_ALARM_REG = alarm_calendar_to_bcd(timeptr);
+
+		rtc_status_reg = RTC->RTC_STATUS_REG;
+		if (!(rtc_status_reg & RTC_RTC_STATUS_REG_RTC_VALID_CAL_ALM_Msk) ||
+			!(rtc_status_reg & RTC_RTC_STATUS_REG_RTC_VALID_TIME_ALM_Msk)) {
+			RTC->RTC_TIME_ALARM_REG = rtc_time_alarm_reg;
+			RTC->RTC_CALENDAR_ALARM_REG = rtc_calendar_alarm_reg;
+			RTC->RTC_ALARM_ENABLE_REG = rtc_alarm_enable_reg;
+			ret = -EINVAL;
+		} else {
+			RTC->RTC_ALARM_ENABLE_REG = tm_to_rtc_alarm_mask(mask);
+		}
+
+		RTC->RTC_INTERRUPT_ENABLE_REG = RTC_RTC_INTERRUPT_ENABLE_REG_RTC_ALRM_INT_EN_Msk;
+	}
+
+	k_mutex_unlock(&data->lock);
+
+	return ret;
+}
+
+static int rtc_smartbond_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+						struct rtc_time *timeptr)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	if ((timeptr == NULL) || (mask == NULL)) {
+		LOG_ERR("No pointer is provided to store the requested alarm time/mask");
+		return -EINVAL;
+	}
+
+	if (!data->is_rtc_configured) {
+		LOG_WRN("RTC is not initialized yet");
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	bcd_to_alarm_calendar(timeptr);
+	bcd_to_alarm_time(timeptr);
+	*mask = rtc_to_tm_alarm_mask(RTC->RTC_ALARM_ENABLE_REG);
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+
+static int rtc_smartbond_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
```

Comment:
move this after the automatic variables, it looks quite out of place right now,  no reason to have it before those